Mathlib/Logic/Lemmas.lean

```diff
@@ -73,3 +73,7 @@ lemma Prop.exists {f : Prop → Prop} : (∃ p, f p) ↔ f True ∨ f False :=
   ⟨fun ⟨p, h⟩ ↦ by refine' (em p).imp _ _ <;> intro H <;> convert h <;> simp [H],
     by rintro (h | h) <;> exact ⟨_, h⟩⟩
 #align Prop.exists Prop.exists
+
+open Function
+
+lemma injective_not : Injective Not := fun _ _ ↦ by simpa using not_iff_not.mp
```

Comment:
The naming guide says this should be called `not_injective`; though I'd be tempted to go further and prove `not_involutive` instead.